src/v/cluster/scheduling/constraints.cc

```diff
@@ -168,13 +168,15 @@ soft_constraint_evaluator distinct_rack(
             for (auto [node_id, shard] : _replicas) {
                 auto rack = _state.get_rack_id(node_id);
                 if (!rack.has_value()) {
```

Comment:
I am wondering if we can have more generic solution i.e. weighted soft constraint. we would assign separate weight parameters to each `soft_constraint_evaluator` this way we could control soft constrains with more flexibility

Reply:
I think that we can build a DAG using soft constraints and then use A* to search a path in that graph to determine a replica placement, or something like this. But it's kind of a big change.

Reply:
yeah, that would be a big change, let's leave it as is

Reply:
nice idea @mmaslankaprv / @Lazin . seems like something we might be able to look into when we start to think about online partition rebalancing, too.